docs/vivisect/devguides/eventsystem.rst

```diff
@@ -0,0 +1,14 @@
+.. _eventsystem:
+
+The Event System
+################
+
+Vivisect's event system is how vivisect informs itself of what's going on, what pieces of analysis should be saved, what remote calls are happening, and just generally what's going on.
+
+More formally, the event system is comprised of a series of handler functions on the VivWorkspaceCore class in `vivisect/base.py`, as well as a firing mechanism (the `_fireEvent` method on the VivWorkspaceCore class). As the main workspace fires events, the handler registered for that event runs. So when something like `VivWorkspace.delLocation` is called, `delLocation` runs and can do whatever setup it wants, and then as a final function call, it calls `self._fireEvent` with the ID of the event it wants to fire (so in this example, `VWE_DELLOCATION`), and a tuple representing the dat it wants to pass to the handler function.
+
+On the surface, this seems like a little bit of overkill. However, this architecture buys us two useful features, along with one limitation. The first thing this buys us is our remote capabilities. Along with several other subsystems, vivisect also brings along a subproject called cobra, which is our own RPC mechanism. Vivisect uses this 
+
+Secondly, the event system is how vivisect saves the workspace. Every time `_fireEvent` is called, the event type and the event info are appended are the running list of events that is a property on the VivWorkspace class.
```

Comment:
iirc, there is one or two exception(s).  i don't recall what they are, but something doesn't get saved, or doesn't get sent across the wire.

Reply:
Are you referring to the `local=True` or `skip` stuff on `_fireEvent`? Or something else? Because neither of those are particularly exercised, nor are they exposed to anything that is really user facing.

Reply:
i'm not remembering right now.  there used to be an event or two which were not transmitted to the server.

Reply:
If there is, I'm not familiar with it. Though I don't really make use of the remote functionality all that much.